src/cryptonote_core/cryptonote_core.cpp

```diff
@@ -944,23 +944,59 @@ namespace cryptonote
     return ret;
   }
   //-----------------------------------------------------------------------------------------------
-  bool core::handle_incoming_txs(const std::vector<tx_blob_entry>& tx_blobs, std::vector<tx_verification_context>& tvc, bool keeped_by_block, bool relayed, bool do_not_relay)
+  bool core::handle_incoming_txs(std::vector<tx_blob_entry> tx_blobs, std::vector<tx_verification_context>& tvc, bool allow_defering, bool keeped_by_block, bool relayed, bool do_not_relay)
   {
     TRY_ENTRY();
     CRITICAL_REGION_LOCAL(m_incoming_tx_lock);
 
-    struct result { bool res; cryptonote::transaction tx; crypto::hash hash; };
-    std::vector<result> results(tx_blobs.size());
+    const size_t prior_txs = m_defered_txs.size();
+
+    for (size_t i = 0; i < tx_blobs.size(); ++i)
+    {
+      // slow linear search, but preserves ordering
+      if (std::find_if(m_defered_txs.rbegin(), m_defered_txs.rend(), [&](const defered_tx_t &dtx) { return dtx.blob.blob == tx_blobs[i].blob; }) == m_defered_txs.rend())
+        m_defered_txs.push_back({std::move(tx_blobs[i]), keeped_by_block, relayed, do_not_relay});
```

Comment:
This implies `m_defered_txs.size() != tx_blobs.size() + prior_txs` in case `tx_blobs` contains some entries already held by `m_defered_txs`, which then breaks the below assertion `tvc_all.size() == tx_blobs.size() + prior_txs`.

Therefore it seems better to me to turn this `if` into an assertion. Or is there a real need to support such cases of overlaps between `tx_blobs` and `m_defered_txs`?

Reply:
That'd still allow the case where the defer threshold is reached, I'll think some more.

Reply:
I pushed a proposed change.